src/Our.Umbraco.Vorto/Web/UI/App_Plugins/Vorto/js/vorto.js

```diff
@@ -318,77 +319,107 @@
+
+            var languagesKey = 'getLanguages_' + currentSection + '_' + editorState.current.id + '_' + editorState.current.parentId + '_' + dataType2.guid;
+            var languagesPromise = cacheService.get(languagesKey);
+            if (!languagesPromise) {
+                languagesPromise = vortoResources.getLanguages(currentSection,
+                    editorState.current.id,
+                    editorState.current.parentId,
+                    dataType2.guid);
+                cacheService.set(languagesKey, languagesPromise);
+            }
 
-                // Load the languages (this will trigger everything else to bind)
-                vortoResources.getLanguages(currentSection, editorState.current.id, editorState.current.parentId, dataType2.guid)
-                    .then(function (languages) {
+            // Load the languages (this will trigger everything else to bind)
+            languagesPromise.then(function(languages) {
+                setLanguages(languages);
+            });
+        }
 
-                        $scope.languages = languages;
+        var setDataType = function(dataType) {
+            // Stash the config in scope for reuse
+            $scope.property.config = dataType.preValues;
 
-                        if (!$scope.model.value.values) {
-                            $scope.model.value.values = {};
-                        }
+            // Get the view path
+            $scope.property.viewPath = umbPropEditorHelper.getViewPath(dataType.view);
 
-                        _.each($scope.languages, function (language) {
-                            if (!$scope.model.value.values.hasOwnProperty(language.isoCode)) {
-                                $scope.model.value.values[language.isoCode] = $scope.model.value.values[language.isoCode];
-                            }
-                        });
+            // Get the property alias
+            var propAlias = $scope.model.propertyAlias || $scope.model.alias;
 
-                        $scope.currentLanguage = $scope.activeLanguage = _.find(languages, function (itm) {
-                            return itm.isDefault;
-                        });
+            // Get the content type alias
+            var contentTypeAlias = nodeContext.contentTypeAlias || nodeContext.alias;
 
-                        reSync();
+            // Work out what section we are in
+            var currentSection = getCurrentSection();
 
-                        validateProperty();
+            var dataTypeByAliasKey = 'getDataTypeByAlias_' + currentSection + '_' + contentTypeAlias + '_' + propAlias;
+            var dataTypeByAliasPromise = cacheService.get(dataTypeByAliasKey);
+            if (!dataTypeByAliasPromise) {
+                dataTypeByAliasPromise = vortoResources.getDataTypeByAlias(currentSection, contentTypeAlias, propAlias);
+                cacheService.set(dataTypeByAliasKey, dataTypeByAliasPromise);
```

Comment:
Same as the comment below

Reply:
I have refactored into more meaningful methods